packages/expo-image/src/ExpoImage.web.tsx

```diff
@@ -201,27 +207,49 @@ function getFetchPriorityFromImagePriority(priority: ImageNativeProps['priority'
   return priority && ['low', 'high'].includes(priority) ? priority : 'auto';
 }
 
-export default function ExpoImage({
+function Image({
   source,
-  placeholder,
-  contentFit,
+  events,
   contentPosition,
-  onLoad,
-  transition,
-  onError,
-  responsivePolicy,
-  onLoadEnd,
   priority,
-  ...props
-}: ImageNativeProps) {
-  const { aspectRatio, backgroundColor, transform, borderColor, ...style } = props.style ?? {};
-  const [state, handlers] = useImageState(source);
-  const { placeholder: placeholderStyle, image: imageStyle } = useTransition(transition, state);
-
-  const { containerRef, source: selectedSource } = useSourceSelection(source, responsivePolicy);
+  style,
+}: {
+  source?: ImageSource | null;
+  events?: {
+    onLoad: (((event: React.SyntheticEvent<HTMLImageElement, Event>) => void) | undefined)[];
+    onError: ((({ source }: { source: ImageSource | null }) => void) | undefined)[];
+  };
+  contentPosition?: ImageContentPositionObject;
+  blurhashContentPosition?: ImageContentPositionObject;
+  priority?: string | null;
+  style: React.CSSProperties;
+  blurhashStyle?: React.CSSProperties;
+}) {
+  const objectPosition = getObjectPositionFromContentPositionObject(contentPosition);
+  const uri = source?.uri;
+  return (
+    <img
+      src={uri || undefined}
+      style={{
+        width: '100%',
+        height: '100%',
+        position: 'absolute',
+        left: 0,
+        right: 0,
+        objectPosition,
+        ...style,
+      }}
+      // @ts-ignore
+      // eslint-disable-next-line react/no-unknown-property
+      fetchpriority={getFetchPriorityFromImagePriority(priority || 'normal')}
+      onLoad={(event) => events?.onLoad.forEach((e) => e?.(event))}
+      onError={() => events?.onError.forEach((e) => e?.({ source: source || null }))}
```

Comment:
```suggestion
      fetchpriority={getFetchPriorityFromImagePriority(priority)}
      onLoad={(event) => events?.onLoad.forEach((e) => e?.(event))}
      onError={() => events?.onError.forEach((e) => e?.({ source }))}
```

It would be neat to move those fallbacks/defaults to the props spread at the params definition.